### DISCUSSION TITLE
Multiple places in the codebase only check the first ColumnAttribute

### DISCUSSION THREAD

**Discussion by FakeDomi** [linq2db/linq2db#4705] [General]
### Describe your issue
Despite `ColumnAttribute` having `AllowMultiple = true` set, at least two places in the codebase only check for the first attribute on a property.  
I ran into this issue since I am programmatically attaching attributes with an `IMetadataReader`, but you can reproduce this by just placing two `ColumnAttribute`s on a property.

`ColumnAttribute`'s `SkipOnEntityFetch = true` is only honored when it's present in the first attribute.

Setting the `columnName` in the second attribute works fine for builtin queries, but when using custom SQL expressions, the mapper doesn't see it and falls back to the default name.

**Workaround:** never add more than a single attribute to a property and make the `IMetadataReader` modify the existing attribute, only adding one if it's not present yet.

### Steps to reproduce

Have multiple `ColumnAttribute`s. In the example below, `assetColumnNotSkipped` will have `Manufacturer` set, and an exception will be thrown for `assetsMappingFailed` since the database column has a different name from the property.

```c#
using LinqToDB.Data;
using LinqToDB;
using NpgsqlTypes;
using LinqToDB.Mapping;
using LinqToDB.DataProvider.PostgreSQL;

namespace ReproSample
{
    internal class Program
    {
        private static void Main()
        {
            using DbConnection db = new DbConnection(new DataOptions()
                .UsePostgreSQL("Host=localhost;Username=postgres;Password=debug;Database=assets", PostgreSQLVersion.v15));

            AssetView assetColumnNotSkipped = db.AssetView.First();
            List<AssetView> assetsMappingFailed = db.AssetView.Where(asset => asset.SearchVector.Matches(Functions.ToTsQuery("sam:*"))).ToList();
        }
    }

    internal class DbConnection(DataOptions options) : DataConnection(options)
    {
        public ITable<AssetView> AssetView => this.GetTable<AssetView>();
    }

    [Table("asset_view")]
    public class AssetView
    {
        [Column("manufacturer")]
        [Column(SkipOnEntityFetch = true)]
        public string? Manufacturer { get; set; } = null!;

        [Column(SkipOnEntityFetch = true)]
        [Column("search_vector")]
        public NpgsqlTsVector SearchVector { get; set; } = null!;
    }

    public static class Functions
    {
        [Sql.Expression("to_tsquery('english', {0})", ServerSideOnly = true)]
        public static NpgsqlTsQuery ToTsQuery(string queryText)
        {
            throw new NotImplementedException("The method is only to be called on the server.");
        }

        [Sql.Expression("{0} @@ {1}", ServerSideOnly = true)]
        public static bool Matches(this NpgsqlTsVector vector, NpgsqlTsQuery query)
        {
            throw new NotImplementedException("The method is only to be called on the server.");
        }
    }
}
```

Exception:
```
Npgsql.PostgresException
  HResult=0x80004005
  Message=42703: Spalte asset.SearchVector existiert nicht (Column asset.SearchVector does not exist)

POSITION: 338
  Source=Npgsql
  StackTrace:
   at Npgsql.Internal.NpgsqlConnector.<ReadMessageLong>d__233.MoveNext()
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at Npgsql.NpgsqlDataReader.<NextResult>d__52.MoveNext()
   at Npgsql.NpgsqlDataReader.<NextResult>d__52.MoveNext()
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.<ExecuteReader>d__120.MoveNext()
   at Npgsql.NpgsqlCommand.<ExecuteReader>d__120.MoveNext()
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at LinqToDB.Data.DataConnection.ExecuteReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteDataReader(CommandBehavior commandBehavior)
   at LinqToDB.Data.DataConnection.ExecuteReader()
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReader()
   at LinqToDB.Linq.QueryRunner.BasicResultEnumerable`1.<GetEnumerator>d__8.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Assets.Program.Main() in C:\Repos\Assets\Program.cs:line 22
```

### Environment details

`Linq To DB` version: both 6.0.0-preview.1 and 5.4.1

`linq2db.EntityFrameworkCore` version (if used): none

`Entity Framework Core` provider with version (if used): none

Database (with version): PostgreSQL 17

ADO.NET Provider (with version): none

Operating system: Windows 10

.NET Version: 8.0.10

**Comment by MaceWindu** [comment]
This is expected behavior.

Most of mapping attributes not merged and must declare all configuration options using single attribute instance.

`AllowMultiple=true` used for different goal: to support multi-configuration mappings, e.g. when same table mapping used for different databases or database providers to specify difference in mappings using `Configuration` property on atttribute. When multiple attributes of same type available we select one with most suitable `Configuration` value.

**Comment by FakeDomi** [comment]
Thank you for the explanation. That makes sense - I have misunderstood the meaning of `AllowMultiple=true` for this attribute.
